test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,227 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+	if err != nil {
+		return nil, fmt.Errorf("daemonset %s Create API error: %v", spec.Name, err)
+	}
+	return ds, nil
+}
+
+type iperf2NodeResults struct {
+	Intervals []*IPerfResult
+	Total     *IPerfResult
+}
+
+type iperf2Results struct {
+	ServerNode string
+	Results    map[string]*iperf2NodeResults
+}
+
+var _ = SIGDescribe("Networking IPerf2 [Experimental] [Slow] [Feature:Networking-Performance]", func() {
+	// this test runs iperf2: one pod as a server, and a daemonset of clients
+	f := framework.NewDefaultFramework("network-perf")
+
+	ginkgo.It(fmt.Sprintf("should run iperf2"), func() {
+		nodeList, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err)
+
+		serverPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-server",
+		}
+
+		serverDeployment, err := iperf2ServerDeployment(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server deployment")
+
+		_, err = iperf2ServerService(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server service")
+
+		_, err = iperf2ClientDaemonSet(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 client daemonset")
+
+		clientPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-client",
+		}
+
+		framework.ExpectNoError(e2edeployment.WaitForDeploymentComplete(f.ClientSet, serverDeployment), "unable to wait for server deployment")
+
+		err = wait.PollImmediate(2*time.Second, 100*time.Second, func() (done bool, err error) {
+			clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+			if err != nil {
+				return false, err
+			}
+			if len(clientPodList.Items) < len(nodeList.Items) {
+				return false, nil
+			}
+			for _, pod := range clientPodList.Items {
+				if pod.Status.Phase != v1.PodRunning {
+					return false, nil
+				}
+			}
+			return true, nil
+		})
+		framework.ExpectNoError(err, "unable to wait for client pods to come up")
+
+		serverPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), serverPodsListOptions)
+		framework.ExpectNoError(err)
+		if len(serverPodList.Items) != 1 {
+			framework.Failf("expected 1 server pod, found %d", len(serverPodList.Items))
+		}
+		serverPod := serverPodList.Items[0]
+
+		clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+		framework.ExpectNoError(err)
+		framework.Logf("found %d matching client pods", len(clientPodList.Items))
+		if len(clientPodList.Items) == 0 {
+			framework.Failf("found 0 matching client pods")
+		}
+
+		nodeResults := &iperf2Results{
+			ServerNode: serverPod.Spec.NodeName,
+			Results:    map[string]*iperf2NodeResults{},
+		}
+		for _, pod := range clientPodList.Items {
+			podName := pod.Name
+			hostIP := pod.Status.HostIP
+
+			command := fmt.Sprintf(`iperf -e -p 6789 --reportstyle C -i 1 -c iperf2-server -T "Client on %s"`, hostIP)
```

Comment:
assuming 6789 is a random port, why this one?  worth a note

Reply:
👍